test/Libraries/DynamoPythonTests/PythonEvalTests.cs

```diff
@@ -170,30 +170,46 @@ public void CPythonEngineIncludesTraceBack()
         }
 
         [Test]
-        public void UnsupportedOutputShouldFailGracefullyInCPython()
+        public void OutputPythonObjectDoesNotThrow()
         {
             var code = @"
 import weakref
 
 class myobj:
-  pass
+    def __str__(self):
+        return 'I am a myobj'
 
 o = myobj()
 wr = weakref.ref(o)
 
 OUT = wr
 ";
             var empty = new ArrayList();
-
-            try
+            Assert.DoesNotThrow(() =>
             {
-                DSCPython.CPythonEvaluator.EvaluatePythonScript(code, empty, empty);
-                Assert.Fail("An exception was expected");
-            }
-            catch (Exception exc)
+                Assert.IsTrue(DSCPython.CPythonEvaluator.EvaluatePythonScript(code, empty, empty).ToString().Contains("weakref at"));
```

Comment:
Is `ToString` expected to return the output from `__str__`?

Reply:
yes, this is what ironPython does if `toString` is not defined, which it would not be for python objects.

Reply:
I guess, I'm trying to understand where the `weakref at` comes from. I know there's a weak ref being returned but still. Is it supported now?

Reply:
not really, the `weakref` type in python defines a __str__ like `weak ref to blah at xyz` and we see that here.

Reply:
something like this:
https://github.com/python/cpython/blob/master/Lib/weakref.py#L615

Reply:
to help clarify - a consequence of this PR is that all python types that we cannot marshall we return handles for, which are mostly useless except that they show a string rep, and you can pass them to other python nodes.

Reply:
Gotcha!